src/renderer/features/dynamicDerivations/ImportKeysModal/ImportKeysModal.tsx

```diff
@@ -0,0 +1,111 @@
+import { useUnit } from 'effector-react';
+import { useEffect } from 'react';
+
+import { Alert, BaseModal, Button, InfoLink, InputFile, InputHint } from '@renderer/shared/ui';
+import { useI18n } from '@renderer/app/providers';
+import { importKeysModel, TypedImportedDerivation } from '@renderer/entities/dynamicDerivations';
+import { AccountId } from '@renderer/shared/core';
+import { cnTw } from '@renderer/shared/lib/utils';
+// @ts-ignore
+import templateFile from '@renderer/assets/files/dd-template.yaml';
+
+type Props = {
+  isOpen: boolean;
+  rootAccountId: AccountId;
+  existingKeys: TypedImportedDerivation[];
+  onClose: () => void;
+};
+
+export const ImportKeysModal = ({ isOpen, onClose, rootAccountId, existingKeys }: Props) => {
+  const { t } = useI18n();
+  const validationError = useUnit(importKeysModel.$validationError);
+  const successReport = useUnit(importKeysModel.$successReport);
+
+  useEffect(() => {
+    if (rootAccountId && existingKeys) {
+      importKeysModel.events.importStarted({ derivations: existingKeys, root: rootAccountId });
+    }
+  }, [rootAccountId, existingKeys]);
+
+  const handleFileUpload = (file: File) => {
+    const fileReader = new FileReader();
+    fileReader.onloadend = (e) => {
+      if (e.target?.result) {
+        importKeysModel.events.fileUploaded(e.target['result'] as string);
+      }
+    };
+
+    fileReader.readAsText(file);
+  };
+
+  const getReportText = () => {
+    if (!successReport) return;
+    const addedKeys = t('dynamicDerivations.importKeys.report.addedKeys', { count: successReport.addedKeys });
+    const updatedNetworks = t('dynamicDerivations.importKeys.report.updatedNetworks', {
+      count: successReport.updatedNetworks,
+    });
+    const duplicatedKeys = t('dynamicDerivations.importKeys.report.duplicatedKeys', {
+      count: successReport.duplicatedKeys,
+    });
+    const ignoreNetworks = t('dynamicDerivations.importKeys.report.networksIgnored', {
+      count: successReport.ignoredNetworks.length,
+    });
+
+    return `${addedKeys} ${updatedNetworks} ${successReport.duplicatedKeys ? duplicatedKeys : ''} ${
+      successReport.ignoredNetworks.length ? ignoreNetworks : ''
+    }`;
+  };
+
+  const handleContinue = () => {
+    // send merged keys to other model
+  };
+
+  return (
+    <BaseModal isOpen={isOpen} title={t('dynamicDerivations.importKeys.modalTitle')} onClose={onClose}>
+      <div className="flex flex-col gap-y-4 items-start">
+        <InputFile
+          placeholder={t('dynamicDerivations.importKeys.fileInputPlaceholder')}
+          accept=".yaml"
+          className={cnTw('w-full h-[126px]', validationError && 'mb-2', successReport && 'mb-4')}
+          invalid={Boolean(validationError?.error)}
+          onChange={(file) => handleFileUpload(file)}
```

Comment:
This doesn't work does it?
```suggestion
          onChange={handleFileUpload}
```

Reply:
it works. updated it